### ISSUE TITLE
Make `NOTIFICATION_WM_SIZE_CHANGED´ available to child nodes of the window

### ISSUE THREAD

**Issue by Sauermann** [godotengine/godot-proposals#8788]
### Describe the project you are working on

Godot

### Describe the problem or limitation you are having in your project

Accessing the notification, when the size of the root window changes is very unintuitive.
This problem has been discussed in the bug report https://github.com/godotengine/godot/issues/79336 which was solved by improving the documentation. However the unintuitive way could be improved.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The problem itself can be solved by making `NOTIFICATION_WM_SIZE_CHANGED` available to child nodes of the window. That allows users to easily access when the root window is changed.
Since this change could have a negative performance impact, the performance impact needs to be evaluated beforehand.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Implementation is likely straightforward.
Whenever a `Window` receives a `NOTIFICATION_WM_SIZE_CHANGED`, it automatically sends the notification recursively to child nodes with the exception of other `Window`-nodes.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

In order to access the notifcation in the root window, one has to add a GDScript to the root node, that listens to notifications. A demo project of how to do that is available in https://github.com/godotengine/godot/issues/79336#issuecomment-1631627181.

### Is there a reason why this should be core and not an add-on in the asset library?

This is a core behavior

**Comment by wgetJane** [comment]
i think that emitting a signal would be more straightforward to implement and doesn't require handling the case of having another window as a child node

**Comment by kubasobon** [comment]
I agree with @wgetJane and @Sauermann, so I started https://github.com/godotengine/godot/pull/87210 to present sample implementation and talk implementation details.

The fact that the only way to get consistently notified about window being resized is to set `/root` object script is not ideal.

**Comment by ghuser404** [comment]
`NOTIFICATION_WM_POSITION_CHANGED` was added in September (without the signal though). The `NOTIFICATION_WM_SIZE_CHANGED` needs to happen.

Also, I'm not sure what stance is on notifications vs signals, but if there are no plans to deprecate one or the other and Godot intends to keep using both, then both `NOTIFICATION_WM_SIZE_CHANGED` notification and `window_size_changed` signal should be added and propagated to children.

This seems like a very fundamental part of an app, and should just be there.
